api/app/Http/Controllers/WorkspaceController.php

```diff
@@ -24,12 +25,34 @@ public function index()
 
     public function saveCustomDomain(CustomDomainRequest $request)
     {
+        if (!$request->workspace->is_pro) {
+            return $this->error([
+                'message' => 'You need Pro plan for use this feature.',
+            ], 403);
+        }
+
         $request->workspace->custom_domains = $request->customDomains;
         $request->workspace->save();
 
         return new WorkspaceResource($request->workspace);
     }
 
+    public function saveEmailSettings(EmailSettingsRequest $request)
+    {
+        if (!$request->workspace->is_pro) {
+            return $this->error([
+                'message' => 'You need Pro plan for use this feature.',
+            ], 403);
+        }
+
+        $settings = $request->workspace->settings;
+        $settings['email_settings'] = $request->all();
+        $request->workspace->settings = $settings;
```

Comment:
This can be a 1 liner:
Use update method, and array merge on request current workspace and new value